Sometimes ```DownloadTest``` failing in CI.

**Describe the bug**
* Sometimes test are failing with view not found exception due to lag in emulator.
* Sometimes it fails due to action mode visible in ```local library fragment``` [More information here](https://github.com/kiwix/kiwix-android/issues/3197)

**Logs**
```
org.kiwix.kiwixmobile.download.DownloadTest > downloadTest[test(AVD) - 11] FAILED 
	java.lang.AssertionError: Couldn't find downloaded file ' Off the Grid ' Original Exception: Off the Grid downloaded successfully. But failed to delete Off the Grid file Actual exception No view found with Id org.kiwix.kiwixmobile:id/zim_file_delete_item
	at org.junit.Assert.fail(Assert.java:89)
```

**Why test failing**

* This is because when we deleting downloaded Zim file sometimes Espresso failed to find ```Delete dialog``` immediately that's why test failure occurs.
* Sometimes no view found(```downloadsFragment```) error occurs in download test.
* Sometimes no view found(```zim_file_delete_item```) error occurs in download test.

--- a/app/src/androidTest/java/org/kiwix/kiwixmobile/download/DownloadRobot.kt
+++ b/app/src/androidTest/java/org/kiwix/kiwixmobile/download/DownloadRobot.kt
@@ -52,18 +52,36 @@ class DownloadRobot : BaseRobot() {
     clickOn(ViewId(R.id.downloadsFragment))
   }
 
+  private fun longClickOnZimFile() {
+    longClickOn(Text(zimFileTitle))
+  }
+
+  private fun clickOnFileDeleteIcon() {
+    clickOn(ViewId(R.id.zim_file_delete_item))
+  }
+
+  private fun assertDeleteDialogDisplayed() {
+    onView(withText("DELETE")).check(matches(isDisplayed()))
+  }
+
+  private fun clickOnDeleteZimFile() {
+    onView(withText("DELETE")).perform(click())
+  }
+
   fun deleteZimIfExists(shouldDeleteZimFile: Boolean) {
     try {
-      longClickOn(Text(zimFileTitle))
-      clickOn(ViewId(R.id.zim_file_delete_item))
-      BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS.toLong())
-      onView(withText("DELETE")).check(matches(isDisplayed()))
-      onView(withText("DELETE")).perform(click())
+      longClickOnZimFile()
+      clickOnFileDeleteIcon()
+      assertDeleteDialogDisplayed()
+      BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS_FOR_ESPRESSO.toLong())
+      clickOnDeleteZimFile()
+      BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS_FOR_ESPRESSO.toLong())
     } catch (e: Exception) {
       if (shouldDeleteZimFile) {
         throw Exception(
           "$zimFileTitle downloaded successfully. " +
-            "But failed to delete $zimFileTitle file"
+            "But failed to delete $zimFileTitle file " +
+            "Actual exception ${e.localizedMessage}"
         )
       }
       Log.i(

--- a/app/src/androidTest/java/org/kiwix/kiwixmobile/initial/download/InitialDownloadRobot.kt
+++ b/app/src/androidTest/java/org/kiwix/kiwixmobile/initial/download/InitialDownloadRobot.kt
@@ -57,13 +57,30 @@ class InitialDownloadRobot : BaseRobot() {
     isVisible(ViewId(R.id.libraryList))
   }
 
+  private fun longClickOnZimFile() {
+    longClickOn(Text(zimFileTitle))
+  }
+
+  private fun clickOnFileDeleteIcon() {
+    clickOn(ViewId(R.id.zim_file_delete_item))
+  }
+
+  private fun assertDeleteDialogDisplayed() {
+    onView(withText("DELETE")).check(matches(isDisplayed()))
+  }
+
+  private fun clickOnDeleteZimFile() {
+    onView(withText("DELETE")).perform(click())
+  }
+
   fun deleteZimIfExists() {
     try {
-      longClickOn(Text(zimFileTitle))
-      clickOn(ViewId(R.id.zim_file_delete_item))
-      BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS.toLong())
-      onView(withText("DELETE")).check(matches(isDisplayed()))
-      onView(withText("DELETE")).perform(click())
+      longClickOnZimFile()
+      clickOnFileDeleteIcon()
+      assertDeleteDialogDisplayed()
+      BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS_FOR_ESPRESSO.toLong())
+      clickOnDeleteZimFile()
+      BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS_FOR_ESPRESSO.toLong())
     } catch (e: Exception) {
       Log.i(
         "TEST_DELETE_ZIM",

--- a/app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/TestUtils.kt
+++ b/app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/TestUtils.kt
@@ -48,6 +48,7 @@ object TestUtils {
   private const val TAG = "TESTUTILS"
   @JvmField var TEST_PAUSE_MS = 250
   var TEST_PAUSE_MS_FOR_SEARCH_TEST = 1000
+  var TEST_PAUSE_MS_FOR_ESPRESSO = 3000
   var TEST_PAUSE_MS_FOR_DOWNLOAD_TEST = 10000
   const val RETRY_COUNT_FOR_FLAKY_TEST = 3
 

